src/domain/safe-apps/entities/schemas/safe-app.schema.ts

```diff
@@ -0,0 +1,49 @@
+import { JSONSchemaType } from 'ajv';
+import { SafeApp } from '../../../../routes/safe-apps/entities/safe-app.entity';
+import { SafeAppAccessControl } from '../safe-app-access-control.entity';
+import { SafeAppProvider } from '../safe-app-provider.entity';
+
+export const safeAppProviderSchema: JSONSchemaType<SafeAppProvider> = {
```

Comment:
Out of curiosity, is it possible to move the `nullable` to where we use `$ref`?

Thinking about the use-case where we want to re-use the schema in a non-nullable way.
Usually I try to make the root schema as non-nullable and apply the nullable at the property level.

Wdyt?

Reply:
Yes, it's better, I agree, thanks!

Reply:
Hmm, I tried it and got this error:
```
Error: "nullable" cannot be used without "type"
```

Now I recall that moving `nullable` to the non-root schema was to in order to keep using `$ref` and `JSONSchemaType` (TS type-aware) instead of simply `Schema` (untyped). I couldn't find another way except by moving it, I could let the `nullable` in the root schema like this:
```
provider: { anyOf: [{ type: 'null' }, { $ref: 'safeAppProviderSchema' }] },
```
But that would make us to use `Schema` instead of `JSONSchemaType`. At leat I couldn't find any workaround.

Reply:
I left it as `Schema` and moved the nullable to the root entity in [32a5f1eb7e](https://github.com/5afe/safe-client-gateway-nest/commit/32a5f1eb7eb7e1e2b02d5e785aba5335a72d3cac), please let me know if you think there's a better way 🙂

Reply:
I think this is ok. `JSONSchemaType` doesn't seem to cover some cases that make our schemas more readable. Recently I had an issue with using `discriminator` and `$merge` together.

For these cases I'd say that we can just use `Schema` as long as we correctly validate the datasources 👍 .

The approach that I'm considering is to always start with `JSONSchemaType`. If there's an incompatibility with it with the schema that we are writing then we fall back to `Schema`. Wdyt?

Reply:
For sure, that's the approach I'm following, start with `JSONSchema` until I find some incompatibility, try to fight it in some way, and finally surrender and use `Schema` 🙂

Reply:
> For sure, that's the approach I'm following, start with `JSONSchemaType` until I find some incompatibility, try to fight it in some way, and finally surrender and use `Schema` 🙂

Yes. I agree! 💯